graphql-parser/src/lexer.rs

```diff
@@ -2,14 +2,23 @@
 use std::str::CharIndices;
 use std::iter::Peekable;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(PartialEq, Debug)]
 pub enum Token<'a> {
-  Eof,
+  Eof(usize, usize),
+  Punctuator(Punctuator, usize),
```

Comment:
I really feel like `usize`-`usize` is better for consistency instead of `usize` and another `usize` inside `Spread`. It might make sense if the variables were named, but since its a tuple, its just confusing. Also, have you thought about using Rust's [`Range`](http://doc.rust-lang.org/std/ops/struct.Range.html) type?